datamonster_api/lib/datamonster.py

```diff
@@ -247,3 +256,219 @@ def _avro_to_df(self, avro_buffer):
+            and calls with _pk2ticker=True
+
+        Return the dimensions for this data source, filtered by `filters`.
+
+        :return: a `DimensionSet` object - say, ``dimset` - an iterable through a dimension dicts,
+            filtered as requested. The object has a additional metadata:
+
+                `max_date`:  (string) max of the `max_date`s of the dimension dicts;
+                `min_date`:  (string) min of the `min_date`s of the dimension dicts;
+                `row_count`:  (int) sum of the `row_count`s of the dimension dicts;
+                `len(dimset)`: (int) number of dimension dicts in the collection
+
+            Each dimension dict has these keys:
+            'max_date', 'min_date', 'row_count', 'split_combination'.
+            The first two are dates, as strings in ISO format; `'row_count'` is an int;
+            `'split_combination'` is a dict, containing keys for this datasource --
+            things you can filter for.
+
+            EXAMPLE
+            --------
+            Assuming `dm` is a DataMonster object, and given this datasource and company:
+
+                datasource = next(dm.get_datasources(query='1010data Credit Sales Index'))
+                the_gap = dm.get_company_by_ticker('GPS')
+
+            this call to `get_dimensions`
+
+                dimset = dm.get_dimensions_for_datasource(
+                                datasource,
+                                filters={'section_pk': the_gap.pk,
+                                         'category': 'Banana Republic'})
+
+            returns an iterable, `dimset`, to a collection with just one dimensions dict.
+            The following loop
+
+                for dim in dimset:
+                    pprint(dim)
+
+            prettyprints the single dimension dict:
+
+                {'max_date': '2019-06-21',
+                 'min_date': '2014-01-01',
+                 'row_count': 1998,
+                 'split_combination': {'category': 'Banana Republic',
+                                       'country': 'US',
+                                       'section_pk': 707}}]
+
+        :raises: DataMonsterError if `filters` is not a dict or is not JSON-serializable.
+            Re-raises `DataMonsterError` if self.client.get() raises that.
+        """
+        self._check_param(datasource=datasource)
+
+        params = {}
+        if filters:
+            params['filters'] = self.to_json_checked(filters)
+        if self.SPLITS_PAGESIZE:
+            params['pagesize'] = self.SPLITS_PAGESIZE
+
+        url = self._get_dimensions_path(uuid=datasource.id)
+        if params:
+            url = ''.join([url, '?', six.moves.urllib.parse.urlencode(params)])
+
+        # Let any DataMonsterError from self.client.get() happen -- we don't occlude them
+        return DimensionSet(url, self, _pk2ticker=_pk2ticker)
+
+    def _convert_section_pks_to_tickers(self, dimension, pk2ticker_memos):
```

Comment:
Was it specifically requested that we convert sections to tickers here?  It seems odd that we would provide a split that doesn't actually exist.

Reply:
Also, why don't we just store the pk2ticker map as a class object?  It seems easier to just make it available to an instance of the class.

Reply:
Yes it was specifically requested: "I want to be in ticker space, not section_pk space." (Elizabeth Fuller) Tickers are meaningful & memorable; section_pk's, not.

Reply:
Exposing the pk2ticker map, or its inverse, wouldn't be a bad thing.

Reply:
To ensure that '/' is always used as path separator, even on Windows, one might use `pathlib` (Py3.4 and backported to Py2), which has a base class `PurePath` with an `as_uri()` method, and documented subclasses `PosixPath` and `WindowsPath`.

It wouldn't help here, though.

Reply:
This comment doesn't look likes it's been addressed.  It seems way easier to store this on the class than to keep track of where it's being passed around.

Reply:
"On the class", as an instance attribute not a class attr. To make this easier I moved the two helper methods that passed it around from DataMonster to DimensionSet, where they belong.

It is probably more useful to users to have the map pk2company:
    `pk => dm.get_company_by_id(pk)` 
rather than pk2ticker, whose values are usually tickers but sometimes names, with no way given to tell which one.

In any case, the contents of the map will be dynamic during an iteration, won't be "done" until after a complete iteration - but if clearly doc'd that's not so weird.